### PR TITLE
Move logic from Setup/SelfUpdate in handlers

### PR DESCRIPTION
**Pull request by chalasr**

### CODE REVIEW

File: src/Command/SelfUpdate.php [hunk 1]
```diff
@@ -71,21 +72,33 @@ protected function execute(InputInterface $input, OutputInterface $output)
             return 0;
         }
 
+        $currentBuild = realpath($_SERVER['argv'][0]) ?: $_SERVER['argv'][0];
```
Comment (ogizanagi) [inline src/Command/SelfUpdate.php]: Duplicated code with l.59 (if still relevant)

File: src/Command/SelfUpdate.php [hunk 2]
```diff
@@ -55,10 +56,10 @@ protected function execute(InputInterface $input, OutputInterface $output)
             return 1;
         }
 
-        $this->client = new Client();
+        $handler = new SelfUpdateHandler(realpath($_SERVER['argv'][0]) ?: $_SERVER['argv'][0]);
```
Comment (ogizanagi) [inline src/Command/SelfUpdate.php:59]: Seems it does not require this argument anymore.

File: src/Command/SelfUpdate.php [hunk 3]
```diff
@@ -71,21 +72,33 @@ protected function execute(InputInterface $input, OutputInterface $output)
             return 0;
         }
 
+        $currentBuild = realpath($_SERVER['argv'][0]) ?: $_SERVER['argv'][0];
+
         try {
-            $this->updateBuild($latestTag);
-        } catch (\PharException $e) {
+            $handler->handle(function ($latestBuild) use ($currentBuild) {
```
Comment (ogizanagi) [inline src/Command/SelfUpdate.php]: I wonder if the separation of concerns made here is relevant or if the `SelfUpdate` handler should both request the new phar and replace the old one. I don't see much value having this in the command which is only here to gather inputs from the outside env and user and display informations. This callback does none of these.

File: src/Handler/SelfUpdate.php [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+namespace Manala\Manalize\Handler;
+
+use GuzzleHttp\Client;
+use Manala\Manalize\Application;
+use Manala\Manalize\Exception\HandlingFailureException;
+
+/**
+ * @author Robin Chalas <robin.chalas@gmail.com>
+ */
+class SelfUpdate implements Handler
```
Comment (ogizanagi) [inline src/Handler/SelfUpdate.php]: I wonder if we really need this interface. Actually we're never using it but instead clearly rely on implementations (which is not something bad here).

Comment (chalasr) [reply src/Handler/SelfUpdate.php]: I think you're right

**Comment by chalasr** [comment]
@ogizanagi `Handler` interface has been removed, this should be ready.

**Review by ogizanagi — APPROVED** [review]

**Comment by ogizanagi** [comment]
Great !

**Pull request merged**